### PR TITLE
Update token renew docs

### PR DESCRIPTION
**Pull request by pmmukh**
- Clarifies some details around the `increment` parameter for token renewal

Fixes https://github.com/hashicorp/vault/issues/12556

### CODE REVIEW

File: command/token_renew.go [hunk 1]
```diff
@@ -77,8 +77,8 @@ func (c *TokenRenewCommand) Flags() *FlagSets {
 		Default:    0,
 		EnvVar:     "",
 		Completion: complete.PredictAnything,
-		Usage: "Request a specific increment for renewal. Vault is not required " +
-			"to honor this request. If not supplied, Vault will use the default " +
+		Usage: "Request a specific increment for renewal. Vault will not honor this" +
+			"request for periodic tokens. If not supplied, Vault will use the default " +
```
Comment (mladlow) [inline command/token_renew.go:81]: I have no idea about this, but if we think there might be cases other than periodic tokens where increment isn't honored, we may want to leave this more open and hybridize the statement between "is not required to honor" and restricting the limit to periodic tokens?

Comment (mladlow) [reply command/token_renew.go:81]: Like if there are global maximums that would restrict how long the token could be renewed for?

File: website/content/api-docs/auth/token.mdx [hunk 2]
```diff
@@ -337,8 +337,10 @@ if there is a lease associated with it.
 
 - `token` `(string: <required>)` - Token to renew. This can be part of the URL
   or the body.
-- `increment` `(string: "")` - An optional requested lease increment can be
-  provided. This increment may be ignored.
+- `increment` `(string: "")` - An optional requested increment duration can be
```
Comment (mladlow) [inline website/content/api-docs/auth/token.mdx:340]: I think this is much clearer!

**Review by mladlow — APPROVED** [review]

File: command/token_renew.go (hunk 1 above)
Comment (pmmukh) [reply command/token_renew.go:81]: Aha yes that is absolutely correct, just noticed this after you mentioned it https://github.com/hashicorp/vault/blob/main/sdk/framework/lease.go#L100, I'll wordsmith this a bit more, thanks for the catch!

File: website/content/api-docs/auth/token.mdx (hunk 2 above)
Comment (pmmukh) [reply website/content/api-docs/auth/token.mdx:340]: thanks! 😀

File: command/token_renew.go (hunk 1 above)
Comment (pmmukh) [reply command/token_renew.go:81]: @mladlow how does "This increment may not be honored, for instance in the case of periodic tokens." sound ?

Comment (mladlow) [reply command/token_renew.go:81]: Sounds good to me!

Comment (pmmukh) [reply command/token_renew.go]: 1c8dad2

**Pull request merged**